src/classes/RD_RecurringDonations.cls

```diff
@@ -216,29 +219,41 @@ public class RD_RecurringDonations {
                 //if this is the open set of opps
                 else if (isClosed == false && isWon == null) {
                     rd.npe03__Next_Payment_Date__c = (date)obj.get('MinDate');
-                    updateMap.put(rdid, rd);                                                       	
+                    updateMap.put(rdid, rd);
                 }
                 //if this is the closed set of opps    
                 else if (isClosed == true && isWon == true) {
-                    rd.npe03__Last_Payment_Date__c = (date)obj.get('MaxDate');
+                    rd.npe03__Last_Payment_Date__c = (date) obj.get('MaxDate');
 
                     //convert rollup total to RD's currency
-                    if (RLLP_OppRollup_UTIL.isMultiCurrency())
-                        rd.npe03__Paid_Amount__c = RLLP_OppRollup_UTIL.ConvertFromCorporate((string)rd.get(RDCurrencyField), (decimal)obj.get('Total'));
-                    else
-                        rd.npe03__Paid_Amount__c = (decimal)obj.get('Total');
-
+                    if (RLLP_OppRollup_UTIL.isMultiCurrency()) {
+                        rd.npe03__Paid_Amount__c = RLLP_OppRollup_UTIL.ConvertFromCorporate((string) rd.get(RDCurrencyField), (decimal) obj.get('Total'));
+                    } else {
+                        rd.npe03__Paid_Amount__c = (decimal) obj.get('Total');
+                    }
                     rd.npe03__Total_Paid_Installments__c = (integer)obj.get('oppcount');
-                    updateMap.put(rdid, rd);                        
-                }                   
+                    updateMap.put(rdid, rd);
+                }
+
+                // if no open opportunities remain because forecast months is too short,
+                // use last payment date to project next payment date
+                if(rd.npe03__Last_Payment_Date__c != null) {
```

Comment:
should we only be doing this for open ended RD's?  was there an existing bug for this scenario?  if so, we should refer to the bug number in the pull request issues section.

Reply:
It is only for open-ended, based on the original query at 186. 
There wasn't a bug for it that I'm aware of, but became necessary as a result of the refactor, since we use getStartDate, which returns the Date Established if Next Payment Date is null. Michael and I can discuss with you further if this isn't clear.

Reply:
This query will also include records with a npe03__Open_Ended_Status__c value of "None" as well as "Open" (since the query is for `!= 'Closed'`). Would be worth understanding the impact

Reply:
Updated ApexDocs header to clarify that this *does* apply to both
    * non-closed (still open) Open-Ended Recurring Donations (status = "Open")
    * and Fixed Length Recurring Donations (status = None or null).